website/server/controllers/api-v3/tasks.js

```diff
@@ -629,7 +642,7 @@ api.addTagToTask = {
     if (validationErrors) throw validationErrors;
 
     let taskId = req.params.taskId;
-    let task = await Tasks.Task.findByIdOrAlias(taskId, user._id, { userId: user._id });
```

Comment:
Don't we check permissions outside of the task query? And, either way, is the better practice to leave the user id blank for group and challenge tasks? Because I check for group permission below.